lib/explorer/backend/lazy_frame.ex

```diff
@@ -0,0 +1,55 @@
+defmodule Explorer.Backend.LazyFrame do
+  @moduledoc """
+  Represents a dataframe in an opaque lazy format.
+  """
+
+  alias Explorer.Backend
+  alias Explorer.Backend.LazySeries
+
+  defstruct dtypes: %{}, names: [], original: nil
+
+  @type t :: %__MODULE__{
+          dtypes: Backend.DataFrame.dtypes(),
+          names: Backend.DataFrame.column_name()
+        }
+  @behaviour Backend.DataFrame
+
+  @doc false
+  def new(df) do
+    %__MODULE__{names: df.names, dtypes: df.dtypes, original: df}
+  end
+
+  @impl true
+  def lazy, do: __MODULE__
+
+  @impl true
+  def to_lazy(ldf), do: ldf
+
+  @impl true
+  def inspect(ldf, opts) do
+    Backend.DataFrame.inspect(ldf.data.original, "OpaqueLazyFrame", nil, opts)
+  end
+
+  @impl true
+  def pull(df, column) do
+    dtype_for_column = df.dtypes[column]
+    data = LazySeries.new(:column, [column])
+    Backend.Series.new(data, dtype_for_column)
+  end
+
+  # TODO: Make the functions of non-implemented functions
+  # explicit once the lazy interface is ready.
```

Comment:
I don't think this TODO applies. We literally just want to raise for everything that is not a pull:

```suggestion
```